drivers/storage/portworx/components.go

```diff
@@ -206,6 +221,25 @@ func (p *portworx) setupLighthouse(t *template) error {
 	return nil
 }
 
+func (p *portworx) setupMonitoring(t *template) error {
+	ownerRef := metav1.NewControllerRef(t.cluster, controllerKind)
+	if err := p.createServiceMonitor(t.cluster, ownerRef); metaerrors.IsNoMatchError(err) {
+		p.warningEvent(t.cluster, failedComponentReason,
+			fmt.Sprintf("Failed to create ServiceMonitor for Portworx. Ensure Prometheus is deployed correctly. %v", err))
```

Comment:
Do we detect if they have prometheus/operator installed at all? this would only fail if the CRD is not registered which means the operator isn't installed. Prometheus itself has very little to do with the service monitor not deploying. If we can detect that they have an operator installed at all we could maybe enhance the ui around enabling monitoring.

Reply:
This will check if there is a kind for prometheus objects. If the object is not found, then we just raise a Warning event and don't fail the operator reconcile loop. We don't need to worry if prometheus operator is running as long as the CRD is present.